crates/types/src/config/cli_option_overrides.rs

```diff
@@ -74,10 +75,21 @@ pub struct CommonOptionCliOverride {
     #[clap(long, env = "RESTATE_CLUSTER_NAME", global = true)]
     pub cluster_name: Option<String>,
 
-    /// If true, then a new cluster is bootstrapped. This node *must* have an admin
-    /// role and a new nodes configuration will be created that includes this node.
-    #[clap(long, global = true)]
-    pub allow_bootstrap: Option<bool>,
+    /// Auto Cluster Provisioning
+    ///
+    /// If true, then this node is allowed to automatically provision as a new cluster.
+    /// This node *must* have an admin role and a new nodes configuration will be created that includes this node.
+    ///
+    /// auto-provision is allowed by default in development mode and is disabled if restate-server runs with `--production` flag
+    /// to prevent cluster nodes from forming their own clusters, rather than forming a single cluster.
+    ///
+    /// Use `restatectl` to provision the cluster/node if automatic provisioning is disabled.
+    ///
+    /// This can also be explicitly disabled by setting this value to false.
+    ///
+    /// Default: true
+    #[clap(long, global = true, alias = "allow_bootstrap")]
```

Comment:
Should this not be?

```suggestion
    #[clap(long, global = true, alias = "allow-bootstrap")]
```

Other compound arguments use kebab-case:*

```
Usage: restate-server <--roles [<ROLES>...]|--node-name <NODE_NAME>|--location <LOCATION>|--force-node-id <FORCE_NODE_ID>|--cluster-name <CLUSTER_NAME>|--auto-provision <AUTO_PROVISION>|--base-dir <BASE_DIR>|--metadata-store-address <METADATA_STORE_ADDRESS>|--bind-address <BIND_ADDRESS>|--advertise-address <ADVERTISE_ADDRESS>|--default-num-partitions <DEFAULT_NUM_PARTITIONS>|--shutdown-timeout <SHUTDOWN_TIMEOUT>|--tracing-endpoint <TRACING_ENDPOINT>|--tracing-runtime-endpoint <TRACING_RUNTIME_ENDPOINT>|--tracing-services-endpoint <TRACING_SERVICES_ENDPOINT>|--tracing-json-path <TRACING_JSON_PATH>|--tracing-filter <TRACING_FILTER>|--log-filter <LOG_FILTER>|--log-format <LOG_FORMAT>|--log-disable-ansi-codes <LOG_DISABLE_ANSI_CODES>>
```